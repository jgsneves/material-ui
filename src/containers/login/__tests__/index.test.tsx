import { fireEvent, render, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Login from '..';
import MockedAuthProvider from '../../../mocks/mockedAuthProvider';

describe('Content Container test suite', () => {
  it('should render proper snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <MockedAuthProvider hash="">
          <Login />
        </MockedAuthProvider>
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('bad user should not be allowed to access the app', async () => {
    const { getByRole, getByLabelText, getByText } = render(
      <BrowserRouter>
        <MockedAuthProvider hash="">
          <Login />
        </MockedAuthProvider>
      </BrowserRouter>,
    );
    const emailInput = getByRole('textbox', {
      name: /email/i,
    });
    const passwordInput = getByLabelText(/senha/i);
    const submitButton = getByRole('button', {
      name: /entrar/i,
    });

    fireEvent.change(emailInput, { target: { value: 'bla' } });
    fireEvent.change(passwordInput, { target: { value: 'pass' } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const errorMessage = getByText(
        'Houve algum erro. Mensagem: Request failed with status code 403',
      );
      expect(errorMessage).toBeInTheDocument();
    });
  });

  it('good user should be allowed to access the app', async () => {
    const { getByRole, getByLabelText, queryAllByText } = render(
      <BrowserRouter>
        <MockedAuthProvider hash="">
          <Login />
        </MockedAuthProvider>
      </BrowserRouter>,
    );
    const emailInput = getByRole('textbox', {
      name: /email/i,
    });
    const passwordInput = getByLabelText(/senha/i);
    const submitButton = getByRole('button', {
      name: /entrar/i,
    });

    fireEvent.change(emailInput, { target: { value: 'teste@gmail.com' } });
    fireEvent.change(passwordInput, { target: { value: '12345' } });
    fireEvent.click(submitButton);

    await waitFor(() => {
      const errorMessage = queryAllByText(
        'Houve algum erro. Mensagem: Request failed with status code 403',
      );
      expect(errorMessage).toEqual([]);
    });
  });
});
