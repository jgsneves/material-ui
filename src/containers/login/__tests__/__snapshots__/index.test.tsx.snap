// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Container test suite should render proper snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1lrpjuv-MuiStack-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Seja bem vindo!
    </h4>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="email"
        id="email-label"
      >
        Email
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="email"
          type="email"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Email
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
        for="password"
        id="password-label"
      >
        Senha
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="password"
          type="password"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-1ftyaf0"
          >
            <span>
              Senha
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Entrar
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
